src/components/SyncServiceSignIn/index.js

```diff
@@ -113,15 +112,10 @@ function GitLab() {
   const defaultProject = 'https://gitlab.com/your/project';
   const [project, setProject] = useState(defaultProject);
   const handleSubmit = (evt) => {
-    const projectId = gitLabProjectIdFromURL(project);
-    if (projectId) {
-      persistField('authenticatedSyncService', 'GitLab');
-      persistField('gitLabProject', projectId);
-      createGitlabOAuth().fetchAuthorizationCode();
-    } else {
-      evt.preventDefault();
-      alert('Project does not appear to be a valid gitlab.com URL');
```

Comment:
@zaz I see that you've refactored the extraction of the `projectId` into the sync backend itself (function `getProject`). Is it still possible to have a validation if no `projectId` can be extracted as here? You've left a TODO here yourself^^

Reply:
Sure. Do you want the alert in `gitLabProjectIdFromURL`? Or would it be possible to move the alert into `createGitlabOAuth`? It might be nice to remove most of the checks in `gitLabProjectIdFromURL` and instead just attempt the OAuth and gracefully handle errors there, unless there would be a downside to that.